pageserver/compaction/src/bin/compaction-simulator.rs

```diff
@@ -0,0 +1,204 @@
+use clap::{Parser, Subcommand};
+use pageserver_compaction::simulator::MockTimeline;
+use rand::Rng;
+use std::sync::OnceLock;
+use std::path::{Path, PathBuf};
+use std::io::Write;
+
+use utils::project_git_version;
+
+project_git_version!(GIT_VERSION);
+
+#[derive(Parser)]
+#[command(
+    version = GIT_VERSION,
+    about = "Neon Pageserver compaction simulator",
+    long_about = "A developer tool to visualize and test compaction"
+)]
+#[command(propagate_version = true)]
+struct CliOpts {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    RunSuite,
+    Simulate(SimulateCmd),
+}
+
+#[derive(Clone, clap::ValueEnum)]
+enum Distribution {
+    Uniform,
+    HotCold,
+}
```

Comment:
I spent some time thinking about how to summarize a workload realistically, but also enable fast simulation. This is what I landed on https://github.com/neondatabase/neon/pull/4411

Reply:
A-ha, I had forgotten about that! Yes, that's a very cool idea. This simulation works by passing around "MockRecord" structs, one for each simulated WAL record. And the mock delta layer creation actually reads through all the MockRecords, copying them to the new layer. The MockRecord only contains the key, LSN and length of the record, but no content. It would indeed be another magnitude faster if we stored just the distribution of the records.